plugins/janus_videoroom.c

```diff
@@ -2771,13 +2789,19 @@ void janus_videoroom_destroy_session(janus_plugin_session *handle, int *error) {
 		if(p)
 			janus_refcount_decrease(&p->ref);
 	} else if(session->participant_type == janus_videoroom_p_type_subscriber) {
+		janus_mutex_lock(&session->mutex);
 		janus_videoroom_subscriber *s = (janus_videoroom_subscriber *)session->participant;
+		if(s)
+			janus_refcount_increase(&s->ref);
 		session->participant = NULL;
+		janus_mutex_unlock(&session->mutex);
 		if(s->room) {
```

Comment:
You're checking ` if(s)` in
https://github.com/meetecho/janus-gateway/pull/2637/files#diff-d96a677f8b8c31f15d44bf3a1d83934e5a6b6bdfab05bd70fd7be6119fb26740R2794
and
https://github.com/meetecho/janus-gateway/pull/2637/files#diff-d96a677f8b8c31f15d44bf3a1d83934e5a6b6bdfab05bd70fd7be6119fb26740R2803

Maybe also change this line to `if(s && s->room) {`?

Reply:
Good point, since we only increase the ref if `s` exists in the first place :+1: